src/Components/LifecycleChart/LifecycleChart.tsx

```diff
@@ -241,11 +241,12 @@ const LifecycleChart: React.FC<LifecycleChartProps> = ({ lifecycleData, viewFilt
   };
 
   const legendNames = React.useMemo(calculateLegendNames, [updatedLifecycleData]);
+  const LEGEND_NAME_PREFIX = '\u00A0\u00A0\u00A0';
```

Comment:
I would add comment there saying something like: `// Nonbreakable space - fix issue with legend hover spacing bug`

Otherwise the PR looks and works nice!